packages/components/src/internal/components/user/UserLink.tsx

```diff
@@ -1,30 +1,90 @@
-import React from 'react';
-import { List } from 'immutable';
-import { User as IUser } from '@labkey/api';
+import React, { FC, useCallback, useState, useEffect } from 'react';
 
 import { User } from '../base/models/User';
-import { AppURL } from '../../url/AppURL';
+import { userCanReadUserDetails } from '../../app/utils';
 
-interface Props {
-    allUsers: List<IUser>;
+import { caseInsensitive } from '../../util/utils';
+
+import { selectRowsUserProps, UserDetailsPanel } from './UserDetailsPanel';
+
+interface UserLinkProps {
     currentUser: User;
-    userId: string;
+    userDisplayValue?: string;
+    userId: number;
 }
 
-export class UserLink extends React.PureComponent<Props> {
-    render() {
-        const { allUsers, currentUser, userId } = this.props;
-        if (!allUsers || !userId) return null;
-
-        let targetUser: IUser = null;
-        if (allUsers) {
-            targetUser = allUsers.find(user => user.userId === parseInt(userId));
-        }
-
-        if (targetUser) {
-            const link = AppURL.create('q', 'core', 'siteusers', userId).toHref();
-            return currentUser.isAdmin ? <a href={link}>{targetUser.displayName}</a> : targetUser.displayName;
-        }
-        return null;
+export const UserLink: FC<UserLinkProps> = props => {
+    const { currentUser, userId, userDisplayValue } = props;
+    const [showDetails, setShowDetails] = useState<boolean>(false);
+    const [targetUserDisplayValue, setTargetUserDisplayValue] = useState<string>();
+    const isSelf = userId === currentUser.id;
+
+    useEffect(() => {
+        (async () => {
+            try {
+                if (!!userId && userDisplayValue === undefined) {
+                    const targetUser = await selectRowsUserProps(userId);
+                    setTargetUserDisplayValue(caseInsensitive(targetUser, 'DisplayName'));
+                } else {
+                    setTargetUserDisplayValue(userDisplayValue);
+                }
+            } catch (error) {
+                console.error(error);
+            }
+        })();
+    }, [userDisplayValue, userId]);
+
+    const toggleDetailsModal = useCallback(() => {
+        setShowDetails(current => !current);
+    }, []);
+
+    if (!userId) return null;
```

Comment:
Seems possible that we pass in a display value and no user id and then we don't ever display anything.

Reply:
Good point. Updated